include/arch/arm/arch/64/mode/object/structures.bf

```diff
@@ -239,19 +239,40 @@ block vm_attributes {
 
 ---- ARM-specific object types
 
--- PGDE, PUDE, PDEs and PTEs, assuming 48-bit physical address
-base 64(48,0)
+block asid_map_none {
+    padding                         63
+    field type                      1
+}
 
--- hw_asids are required in hyp mode
-block pgde_invalid {
-    field stored_hw_asid            8
-    field stored_asid_valid         1
+--- hw_vmids are required in hyp mode
+block asid_map_vspace {
 #ifdef CONFIG_ARM_SMMU
-    field bind_cb                   12
-    padding                         41
+    field bind_cb                   8
+    padding                         8
+#else
+    padding                         16
+#endif
+    field_high vspace_root          36
+#ifdef CONFIG_ARM_HYPERVISOR_SUPPORT
+    padding                         2
+    field stored_hw_vmid            8
```

Comment:
recent processors support 16-bit vmid, so we could just make it 16b-bit.

Reply:
On a processor that supports 16bit VMID we would be able to just use the asid_t value which is also 16bits and not need a stored_hw_vmid slot.

Reply:
right, if we have enough hardware asid, we do not need the mapping. where do we specify that?

Reply:
At the moment all of the supported platforms only have 8bit VMID. If there's a platform that supports 16bit VMID in the future then the kernel would get a config update to make this feature `CONFIG_ARM_HYPERVISOR_SUPPORT && CONFIG_8BIT_VMID`